Source/WebKit/WebProcess/GPU/graphics/RemoteImageBufferProxy.h

```diff
@@ -116,9 +114,13 @@ class RemoteImageBufferProxy : public WebCore::ConcreteImageBuffer<BackendType>
         m_receivedFlushIdentifierChangedCondition.notifyAll();
     }
 
-    void setNeedsFlush(bool needsFlush) final
+    void willUpdate() final
     {
-        m_needsFlush = needsFlush;
+        if (LIKELY(m_hasPendingUpdates))
+            return;
+        if (auto* backend = ensureBackendCreated())
+            backend->ensureNativeImagesHaveCopiedBackingStore();
```

Comment:
> I had to look for this function and I was surprised to see that it just calls the IOSurface fill empty rect hack followed by a call to flushContext().
> Besides flushContext() calls flushDrawingContext() which calls flushDrawingContextAsync(). And flushDrawingContextAsync() returns immediately in this case because m_needsFlush (or m_hasPendingUpdates) is false.

I think your description is not accurate. By the way you write, I think you mistake `ImageBufferIOSurfaceBackend::flushContext()` and `ConcreteImageBuffer::flushContext()`.

`ImageBufferIOSurfaceBackend::flushContext()` does not call `ConcreteImageBuffer::flushDrawingContext()`.